sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedFlux.java

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.paging;
+
+import reactor.core.publisher.Flux;
+
+/**
+ * A contract that represents a Flux that provides the ability to operate on individual items
+ * in pages of type {@link ContinuablePage}, also provide ability to operate on individual
+ * pages.
+ *
+ * @param <O> the type of continuation option for byPage
+ * @param <C> the type of continuation token
+ * @param <T> the type of items in the page
+ * @param <P> the type of page
+ */
+public abstract class ContinuablePagedFlux<O, C, T, P extends ContinuablePage<C, T>> extends Flux<T> {
+    /**
+     * @return a Flux of {@link ContinuablePage} that this Paged Flux represents.
+     */
+    public abstract Flux<P> byPage();
+    /**
+     * @return a Flux of {@link ContinuablePage} identified by the given continuation option.
+     */
+    public abstract Flux<P> byPage(O continuationOption);
```

Comment:
For me this is the main area of contention. The fact that we have the byPage method take `O`, rather than `C`, and that we call this a `continuationOption` rather than a `continuationToken` (because it `O` and not `C`). My preference is to change this class to remove the `O` type, and to instead revert this back to being `C continuationToken`. Subclasses may choose to introduce the `O` type and a different byPage 'overload'. Barring objections from other reviewers (@conniey, @alzimmermsft, etc), this would be my recommendation.

Reply:
totally agree, as you receommended removed the option  generic arg.